File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        vector<long> prefix = {nums[0]};
        for (int i = 1; i < nums.size(); i++) {
            prefix.push_back(prefix.back() + nums[i]);
        }

        int ans = 0;
        for (int i = 0; i < nums.size() - 1; i++) {
            long left_section = prefix[i];
            long right_section = prefix.back() - prefix[i];

            if (left_section >= right_section) {
                ans++;
            }
        }

        return ans;
    }
};

File: 2270-number-of-ways-to-split-array/2270-number-of-ways-to-split-array.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        long prefix[] = new long[nums.length];

        prefix[0] = nums[0];
        for (int i = 1; i < nums.length; i++) {
            prefix[i] = nums[i] + prefix[i - 1];
        }

        int ans = 0;

        for (int i = 0; i < nums.length - 1; i++) {
            if (prefix[i] >= prefix[prefix.length - 1] - prefix[i]) {
                ans++;
            }
        }

        return ans;
    }
}